src/rules/requireReadonlyReactProps.js

```diff
@@ -21,13 +33,16 @@ const isReactComponent = (node) => {
   );
 };
 
-const isReadOnlyObjectType = (node) => {
+const isReadOnlyObjectType = (node, {implicitExactTypes}) => {
   if (!node || node.type !== 'ObjectTypeAnnotation') {
     return false;
   }
 
-  // we consider `{||}` to be ReadOnly since it's exact AND has no props
-  if (node.exact && node.properties.length === 0) {
+  // we consider `{}` to be ReadOnly since it's exact AND has no props (when `implicitExactTypes=true`)
+  // we consider `{||}` to be ReadOnly since it's exact AND has no props (when `implicitExactTypes=false`)
+  if (implicitExactTypes === true && node.exact === false && node.properties.length === 0) {
+    return true;
+  } else if (node.exact === true && node.properties.length === 0) {
     return true;
   }
```

Comment:
I'm wondering if this is nicer and more readable, the close duplication in syntax of your two conditions took me a while to notice that `node.exact` `===` `true/false` in both conditions
```suggestion
  if (node.properties.length === 0) {
    if (node.exact === false && implicitExactTypes === true) {
      return true;
    }
    if (node.exact === true) {
      return true;
    }
  }
```